libbeat/processors/actions/decode_json_fields.go

```diff
@@ -75,7 +80,58 @@ func (f decodeJSONFields) Run(event common.MapStr) (common.MapStr, error) {
 				continue
 			}
 
-			_, err = event.Put(field, output)
+			if f.target != nil {
+				if len(*f.target) > 0 {
+					_, err = event.Put(*f.target, output)
+				} else {
+					switch t := output.(type) {
```

Comment:
The logic looks good. But now the complexity of the `Run` method is a bit high with all of the nested if/for/switch statements. WDYT about breaking into smaller functions?